### DISCUSSION TITLE
how use python_embeded

### DISCUSSION THREAD

**Discussion by extaform** [SWivid/F5-TTS#126] [Q&A]
Hello! How do I use python embedded 3.10 with your application? Let's say the folder with python embedded files is located in the F5-TTS folder after cloning using git. How do I declare an environment variable for your application? Thnx

**Comment by aniketraut16** [comment]
To use Python Embedded 3.10 with an application and set it up with the environment variable pointing to the embedded Python interpreter, follow these steps:

### 1. **Download and Clone Python Embedded 3.10**:
   Assuming you have already cloned the embedded Python files into a folder named `F5-TTS`, make sure the folder contains Python 3.10 binaries (`python.exe` or `pythonw.exe`) and all necessary libraries.

### 2. **Set Environment Variables**:
   You need to declare environment variables to ensure your application uses the Python interpreter located in the `F5-TTS` folder. This can be done by configuring the `PYTHONHOME` and `PYTHONPATH` variables.

#### Windows:

   **Steps to Set Environment Variables in Windows:**
   1. Press `Win + X` and select **System**.
   2. Click **Advanced System Settings** on the left side of the window.
   3. In the **System Properties** window, click on the **Environment Variables** button.
   4. Under **System variables**, click **New** to add a new environment variable.

      - **Variable name**: `PYTHONHOME`
      - **Variable value**: Full path to your `F5-TTS` folder, e.g., `C:\path\to\F5-TTS`

   5. Add another variable for `PYTHONPATH`:
      - **Variable name**: `PYTHONPATH`
      - **Variable value**: `C:\path\to\F5-TTS\Lib` (adjust for your folder structure).

   6. Add the Python embedded folder to the `PATH` variable:
      - Find the `Path` variable under **System variables**, select it, and click **Edit**.
      - Add the full path to the folder containing `python.exe`, e.g., `C:\path\to\F5-TTS`.

   7. Click **OK** to save the changes.

#### Linux/Mac:

   **Steps to Set Environment Variables in Linux or macOS:**
   1. Open the terminal.
   2. Edit your shell configuration file (`.bashrc`, `.zshrc`, or `.bash_profile`) depending on the shell you are using.
   3. Add the following lines to export the environment variables:
   
   ```bash
   export PYTHONHOME=/path/to/F5-TTS
   export PYTHONPATH=/path/to/F5-TTS/Lib
   export PATH=$PYTHONHOME:$PATH
   ```
   4. Save the file and reload your shell using:
   
   ```bash
   source ~/.bashrc    # For bash
   source ~/.zshrc     # For zsh
   ```

### 3. **Test the Setup**:
   To verify that the environment is set up correctly, open a terminal or command prompt and type:
   
   ```bash
   python --version
   ```
   It should display `Python 3.10.x` indicating that the embedded Python is being used.

Now, your application should be able to use Python 3.10 from the `F5-TTS` folder. If your application requires further configuration (like specifying the interpreter path), make sure it points to the `python.exe` located in your `F5-TTS` folder.

**Reply by extaform** [reply]
Hello, thanks for the advice, but that's not what I asked. How do I declare the python 3.10 environment only for the application, and not for the entire operating system?
